src/platforms/UniqueText/UniqueTextCardMenu.tsx

```diff
@@ -10,19 +14,46 @@ type Props = {
 }
 
 const UniqueTextCardMenu = ({ platformGuildId }: Props): JSX.Element => {
-  const { guildPlatforms } = useGuild()
+  const { guildPlatforms, roles } = useGuild()
   const guildPlatform = guildPlatforms?.find(
     (gp) => gp.platformGuildId === platformGuildId
   )
+  const rolePlatform = roles
+    .flatMap((role) => role.rolePlatforms)
+    .find((rp) => rp.guildPlatformId === guildPlatform?.id)
 
   const { isOpen, onOpen, onClose } = useDisclosure()
 
+  const {
+    isOpen: isCapacityTimeOpen,
+    onOpen: onCapacityTimeOpen,
+    onClose: onCapacityTimeClose,
+  } = useDisclosure()
+
+  const toast = useToast()
+
+  const { onSubmit, isLoading } = useEditRolePlatform({
+    rolePlatformId: rolePlatform?.id,
+    onSuccess: () => {
+      toast({
+        status: "success",
+        title: "Successfully updated reward",
+      })
+      onCapacityTimeClose()
+    },
+  })
+
   return (
     <>
       <PlatformCardMenu>
         <MenuItem icon={<PencilSimple />} onClick={onOpen}>
           Edit unique secret
         </MenuItem>
+        {rolePlatform && (
```

Comment:
I thought it might be undefined if someone just adds a guildPlatform with our API endpoint, but actually we should handle this (= disable it) on our backend, so I removed this condition in https://github.com/agoraxyz/guild.xyz/pull/946/commits/d8ae2dabdee818881ed30dfad3c8444c40c8efdb & https://github.com/agoraxyz/guild.xyz/pull/946/commits/5b4132c784ecffc31d27a6b9b69980202aec6e22!